### ISSUE TITLE
MySQL 5.7 and empty/null values

### ISSUE THREAD

**Issue by robbrandt** [zikula/core#3590]
| Q                 | A
| ----------------- | ---
| Zikula Version    | 1.3.12
| PHP Version       | 5.6
| MySQL Version | 5.7.18

### Expected behavior
New users should be able to register without error

### Actual behavior
SQL error.

### Steps to reproduce
This is happening with a brand new Ubuntu 16.04 server instance on Amazon.  It was running for a couple of weeks before I noticed that I wasn't getting any new users.  Attempting to register a new user results in Error in DBUtil::executeSQL: INSERT INTO users (uname,email,pass,passreminder,activated,user_regdate) VALUES ('testnewserver','rbrandt@businessedgesoftware.com','8$HqbC|$5f1ccadda89e873953b9bece2e3d8cde84f5f0e79bb516d22acc247872d42756','car',-32768,'2017-05-19 00:19:55')
SQLSTATE[HY000]: General error: 1364 Field 'ublock' doesn't have a default value

I don't know if this is a good solution or not, but I found that allowing NULL allows the registration to continue.

I found that this was the case for 4 fields in the users table: theme, ublock, tz, & locale.

It seems this is related to a STRICT_MODE in MySQL 5.7.
https://www.percona.com/blog/2016/10/18/upgrading-to-mysql-5-7-beware-of-the-new-strict-mode/

Bottom line, changes MIGHT have to be made in order for applications to run under MySQL's default configuration. Might apply to Zikula 1.4/5 and 2.0 as well.

**Comment by craigh** [comment]
DBUtil is not used in 1.4/5 and completely removed from 2.0. Doctrine is compatible. I'm glad you found a solution for yourself and implemented it.

**Comment by robbrandt** [comment]
It may be that Doctrine takes care of this for 1.4+, but it's not just a DBUtil issue.  It happens any time an insert happens where all columns aren't included, and if some of those columns 1) do not allow NULL, and 2) have no default value.  Put another way, if a table has columns that do not allow NULL and do not have a default value set, those columns must have a specific non-empty value included in the INSERT.

**Comment by craigh** [comment]
> if a table has columns that do not allow NULL and do not have a default value set, those columns must have a specific non-empty value included in the INSERT.

It seems to me that this is the responsibility of the developer, not the core to persist valid values for their entity.

Closing.
